File: Courses/PHP Web Basics - май 2019/16.ExerciseEncapsulationInheritance/01.04.MordorCrueltyPlan/MoodFactory.php
<?php


class MoodFactory
{
    /**
     * @var string
     */
    private $mood;
    public function __construct(int $points)
    {
        $this->setMood($points);
    }
    public function getMood() : string
    {
        return $this->mood;
    }
    public function setMood($points)
    {
        if ($points < -5){
            $this->mood = "Angry";
        }elseif ($points < 0){
            $this->mood = "Sad";
        }elseif ($points < 15){
            $this->mood = "Happy";
        }else{
            $this->mood = "PHP";
        }
    }
}

File: Courses/C# OOP/10. Inheritance Exercise/Exercises_Inheritance/05.MordorCruelPlan/Factories/MoodFactory.cs
using _05.MordorCruelPlan.Moods;
using System;
using System.Collections.Generic;
using System.Text;

namespace _05.MordorCruelPlan.Factories
{
    public class MoodFactory
    {
        public Mood CreateMood(string type)
        {
            type = type.ToLower();

            switch (type)
            {
                case "angry":
                    return new Angry();
                case "sad":
                    return new Sad();
                case "happy":
                    return new Happy();
                case "javascript":
                    return new JavaScript();
                default:
                    throw new Exception("Invalid type");
            }
        }
    }
}
